src/category_theory/arrow_class.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import category_theory.arrow
+
+/-!
+
+# Classes of arrows in a category
+
+If `C` is a category, an `arrow_class C` is a class of arrows in C. It is
+implemented here as `set (arrow C)`.
+
+If `W : arrow_class C`, and `f : X ⟶ Y` is a morphism in `C`, `W f` is the
+condition that `arrow.mk f` belongs to `W`.
+
+-/
+
+noncomputable theory
+
+namespace category_theory
+
+variables (C : Type*) [category C] {D : Type*} [category D]
+
+/-- An `arrow_class C` is a class of arrows in a category `C`. -/
+abbreviation arrow_class := set (arrow C)
+
+instance : has_coe_to_fun (arrow_class C) (λ F, (Π ⦃X Y : C⦄, (X ⟶ Y) → Prop)) :=
+⟨λ F X Y f, arrow.mk f ∈ F⟩
```

Comment:
It's not clear to me why you want this instance. Isn't it more natural to set up a `has_mem` instance?

Reply:
Actually, I do not need this. I put it as a compromise if someone wants to write `W f` instead of `arrow.mk f \in W`.

I did not manage to make `has_mem` isntance work. If I do
```lean
instance {X Y : C} : has_mem (X ⟶ Y) (arrow_class C) := ⟨λ f W, arrow.mk f ∈ W⟩

example (W : arrow_class C) {X Y : C} (f : X ⟶ Y) : true :=
begin
  have foo : f ∈ W := sorry,
end
```
then Lean fails to synthesize type class instance for "has_mem (X ⟶ Y) (arrow_class C)" !?

Reply:
Ok. I have removed the `has_coe_to_fun` instance.